src/hotspot/share/opto/library_call.cpp

```diff
@@ -2652,7 +2655,7 @@ bool LibraryCallKit::inline_unsafe_finish_private_buffer() {
     return false;
   }
   InlineTypeNode* vt = buffer->as_InlineType();
-  if (!vt->is_allocated(&_gvn)) {
+  if (!vt->is_allocated(&_gvn) || VectorSupport::is_vector_payload_mf(vt->inline_klass()->get_InlineKlass())) {
```

Comment:
This will make the `finishPrivateBuffer` un-inlined in all Vector API cases, won't it? This seems not so efficient to me. But I cannot find a better way to fix the masked cases now.

Reply:
You are right, but it will not affect the performance given that code is part of fallback implementation.

Reply:
OK, it's fine to me now. We can revisit this part and fine tune if possible in future. Thanks!